react/Accordion/index.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import Icon from '../Icon'
+import styles from './styles.styl'
+
+class AccordionItem extends Component {
+  constructor(props) {
+    super(props)
+    this.onClick = this.onClick.bind(this)
+    this.state = { selected: false }
+  }
+
+  render() {
+    const { children, label, className } = this.props
+
+    const { selected } = this.state
+
+    const bodyName = label.replace(/\s+/g, '')
+
+    return (
+      <div className={cx(styles['c-accordion-item'], className)}>
+        <div
+          className={cx(
+            styles['c-accordion-title'],
+            {
+              [styles['is-active']]: selected
+            },
+            className
+          )}
+          aria-selected={selected}
+          aria-controls={bodyName}
+          role="tab"
+          tabIndex="0"
+          onClick={this.onClick}
+        >
+          <Icon icon="forward" />
+          {label}
+        </div>
+        <div
+          className={cx(
+            styles['c-accordion-body'],
+            {
+              [styles['is-active']]: selected
+            },
+            className
+          )}
+          aria-hidden={!selected}
+          aria-labelledby={bodyName}
+          role="tabpanel"
+        >
+          {children}
+        </div>
+      </div>
+    )
+  }
+
+  onClick() {
+    this.setState(previousState => ({ selected: !previousState.selected }))
```

Comment:
Move this instruction in a named function and implement `onClick` to execute this behavior:

```js
toggleSelection() {
  this.setState(previousState => ({ selected: !previousState.selected })
}

onClick() {
  this.toggleSelection()
}
```

If it uses more code, it is more explicit on the intention. By the way, you can inline the `onClick` handle to the `props` of the component:

```js
<div onClick={() => this.toggleSelection} >{label}</div>
```

It is way more explicit than `onClick={onClick}`.